docs-ref-services/servicebus.md

```diff
@@ -15,34 +15,189 @@ ms.service: service-bus
+
+As of version 0.50.0 a new AMQP-based API is available for sending and receiving messages. This update involves breaking changes.
+Please read [Migration from v0.21.1 to v0.50.0](#migration-from-v0.21.1-to-v0.50.0) to determine if upgrading is
+right for you at this time.
+
+For documentation on the legacy HTTP-based operations please see [Using HTTP-based operations of the legacy API](#using-http-based-operations-of-the-legacy-api)
+
+
+## Prerequisites
+
+* Azure subscription - [Create a free account](https://azure.microsoft.com/free/)
+* Azure [Service Bus namespace and management credentials](https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-create-namespace-portal)
+* [Python 2.7-3.7](https://www.python.org/downloads/)
+
 
 ## Install the libraries
 ```bash
 pip install azure-servicebus
 ```
 
+## Create Client
+
+To get started, you will need to create a ServiceBusClient using the
+namespace and credentials that can be retrieved from the Azure Portal.
+```python
+import os
+from azure.servicebus import ServiceBusClient
+
+namespace = os.environ['SB_NAMESPACE']
+policy = os.environ['SB_ACCESS_POLICY']
+key = os.environ['SB_ACCESS_KEY']
+
+sb_client = ServiceBusClient(
+    service_namespace=namespace,
+    shared_access_key_name=policy,
+    shared_access_key_value=key)
+```
+A client can also be created with the Connection String from the portal directly:
+```python
+connection_str = os.environ['SB_CONN_STR']
+
+sb_client = ServiceBusClient.from_connection_string(connection_str)
+```
+
+
 ## ServiceBus Queues
 ServiceBus Queues are an alternative to Storage Queues that might be
 useful in scenarios where more advanced messaging features are needed
 (larger message sizes, message ordering, single-operation destructive
 reads, scheduled delivery) using push-style delivery (using long
 polling).
 
-The service can use Shared Access Signature authentication, or ACS
-authentication.
+The service can use Shared Access Signature authentication.
+
+### Create queue
+This creates a new queue within the ServiceBus namespace. If a queue of the same name already exists an error will be raised. 
+```python
+sb_client.create_queue("MyQueue")
+```
+Optional parameters to configure the queue behaviour can also be specified
+```python
+sb_client.create_queue(
+    "MySessionQueue",
+    requires_session=True  # Create a sessionful queue
+    max_delivery_count=5  # Max delivery attempts per message
+)
+```
+
+### Get a queue client
+A QueueClient can be used to send and receive messages from the queue, along with other operations.
+```python
+queue_client = sb_client.get_queue("MyQueue")
+```
+
+### Sending messages
+The queue client can send one or messages at a time:
+```python
+from azure.servicebus import Message
+
+message = Message("Hello World")
+queue_client.send(message)
+
+message_one = Message("First")
+message_two = Message("Second")
+queue_client.send([message_one, message_two])
```

Comment:
Double-checking that send didn't take an *args parameter...

Reply:
Yes - I changed this because of the syntax in Python 2.7 - I can't require that people use keyword args for the optional parameters and it can result in weird behaviour. So I thought it best to be explicit and accept both a Message or an iterable of Messages.

In reality if people are sending multiple messages at once, they likely already have them in some kind of list, so it's actually easier to just pass that in rather then making them add "*".